### PR TITLE
2715-Add availability test in app insight

### PR DESCRIPTION
**Pull request by rizzkhan1**
### Context
Add availability test for BE hostname  and custom hostnames in application insight

### CODE REVIEW

File: azure/template.json [hunk 1]
```diff
@@ -284,7 +291,20 @@
     "databaseResourceNamePrefix": "[toLower(concat(parameters('resourceEnvironmentName'),'-', parameters('databaseServiceName')))]",
     "databaseServerName": "[concat(variables('databaseResourceNamePrefix'), '-psql')]",
     "databaseUsername": "[concat(parameters('databaseUsername'), '@', variables('databaseServerName'))]",
-    "storageAccountName": "[replace(concat(variables('resourceNamePrefix'), 'str'), '-', '')]"
+    "storageAccountName": "[replace(concat(variables('resourceNamePrefix'), 'str'), '-', '')]",
+    "defaultAvailabilityCheckHosts": "[if(greater(length(parameters('customHostName')), 0), createArray(concat('azcheck:', variables('appServiceName'), '.azurewebsites.net/ping'), concat('check:', parameters('customHostName'), '/ping')), createArray(concat('azcheck:', variables('appServiceName'), '.azurewebsites.net/ping')))]",
```
Comment (misaka) [inline azure/template.json]: Should this be:

```suggestion
    "defaultAvailabilityCheckHosts": "[if(greater(length(parameters('customHostName')), 0), createArray(concat('azcheck:', variables('appServiceName'), '.azurewebsites.net/ping'), concat('azcheck:', parameters('customHostName'), '/ping')), createArray(concat('azcheck:', variables('appServiceName'), '.azurewebsites.net/ping')))]",
```

Comment (misaka) [reply azure/template.json]: I found this hard to read, so I've reformatted it. Nothing visibly wrong, it's just way easier to read. :)

```C
if(greater(length(parameters('customHostName')), 0),
   createArray(
     concat('azcheck:', variables('appServiceName'), '.azurewebsites.net/ping'),
     concat('check:', parameters('customHostName'), '/ping')
   ),
   createArray(
     concat('azcheck:', variables('appServiceName'), '.azurewebsites.net/ping')
   )
)
```

File: azure/template.json [hunk 2]
```diff
@@ -284,7 +291,20 @@
...
+    "availabilityCheckHosts": "[if(greater(length(parameters('customAvailabilityMonitors')), 0), concat(variables('defaultAvailabilityCheckHosts'), parameters('customAvailabilityMonitors')), variables('defaultAvailabilityCheckHosts'))]",
```
Comment (misaka) [inline azure/template.json]:
```C
if(greater(length(parameters('customAvailabilityMonitors')), 0),
   concat(
     variables('defaultAvailabilityCheckHosts'),
     parameters('customAvailabilityMonitors')
   ),
   variables('defaultAvailabilityCheckHosts')
)
```

Do these not need the `azcheck:` prefixes?

File: azure/template.json (hunk 1 above)
Comment (rizzkhan1) [reply azure/template.json]: > Should this be:

There are two availability checks, `check`  and `azcheck`.  One for azurewebsite.net address and other for custom hostname.

File: azure/template.json (hunk 2 above)
Comment (rizzkhan1) [reply azure/template.json]: No. `azcheck` is inside defaultAvailabilityCheckHosts variable.  Here we are just checking if custom monitors are provided as a parameter and taking actions accordingly.

**Pull request merged**